mmseg/models/losses/inverseform_loss.py

```diff
@@ -0,0 +1,329 @@
+    """
+
+    def __init__(self,
+                 classes,
+                 weight=None,
+                 reduction='mean',
+                 ignore_index=255,
+                 norm=False,
+                 upper_bound=1.0,
+                 fp16=False,
+                 edge_weight=0.3,
+                 inv_weight=0.3,
+                 seg_weight=1,
+                 att_weight=0.1,
+                 loss_name='loss_inverseform'):
+        super().__init__()
+        self.num_classes = classes
+        self.seg_loss = ImageBasedCrossEntropyLoss2d(
+            classes=classes,
+            weight=weight,
+            reduction=reduction,
+            ignore_index=ignore_index,
+            norm=norm,
+            upper_bound=upper_bound,
+            fp16=fp16).cuda()
+        self.inverse_distance = InverseTransform2D()
+        self.edge_weight = edge_weight
+        self.seg_weight = seg_weight
+        self.att_weight = att_weight
+        self.inv_weight = inv_weight
+        self._loss_name = loss_name
+
+    def bce2d(self, input, target):
+        """
+        To calculate weighted cross-entropy of two binary boundary map
+        input and target are both binary boundary map
+        Args:
+        input: pred binary boundary map
+        target: ground truth binary boundary map generated by using Sobel
+                filter and threshold on ground truth segmentation mask
+        """
+        n, c, h, w = input.size()
+
+        log_p = input.transpose(1, 2).transpose(2, 3).contiguous().view(1, -1)
+        target_t = target.transpose(1,
+                                    2).transpose(2,
+                                                 3).contiguous().view(1, -1)
+        target_trans = target_t.clone()
+
+        pos_index = (target_t == 1)
+        neg_index = (target_t == 0)
+        ignore_index = (target_t > 1)
+
+        target_trans[pos_index] = 1
+        target_trans[neg_index] = 0
+
+        pos_index = pos_index.data.cpu().numpy().astype(bool)
+        neg_index = neg_index.data.cpu().numpy().astype(bool)
+        ignore_index = ignore_index.data.cpu().numpy().astype(bool)
+
+        weight = torch.Tensor(log_p.size()).fill_(0)
+        weight = weight.numpy()
+        pos_num = pos_index.sum()
+        neg_num = neg_index.sum()
+        sum_num = pos_num + neg_num
+        weight[pos_index] = neg_num * 1.0 / sum_num
+        weight[neg_index] = pos_num * 1.0 / sum_num
+
+        weight[ignore_index] = 0
+
+        weight = torch.from_numpy(weight)
+        weight = weight
+        loss = F.binary_cross_entropy_with_logits(
+            log_p, target_t, weight, size_average=True)
+        return loss
+
+    def edge_attention(self, input, target, edge):
+        n, c, h, w = input.size()
+        filler = torch.ones_like(target) * 255
+        return self.seg_loss(input,
+                             torch.where(edge.max(1)[0] > 0.8, target, filler))
+
+    def forward(self, inputs, targets, do_rmi=None):
```

Comment:
Might add `**kwargs` to the parameter list, since the `BaseDecodeHead` need a unify loss interface.
https://github.com/open-mmlab/mmsegmentation/blob/main/mmseg/models/decode_heads/decode_head.py#L334